module/VuFind/src/VuFind/ILS/Driver/CombinedDriver.php

```diff
@@ -0,0 +1,1346 @@
+    }
+
+    /**
+     * Get Consortial Holdings
+     *
+     * This is responsible for retrieving the holding information of a certain
+     * consortial record.
+     *
+     * @param string $id     The record id to retrieve the holdings for
+     * @param array  $patron Patron data
+     * @param array  $ids    The (consortial) source records for the record id
+     *
+     * @return array         On success, an associative array with the following
+     *  keys: id, availability (boolean), status, location, reserve, callnumber,
+     *  duedate, number, barcode.
+     * @throws ILSException
+     * @throws DateException
+     */
+    public function getConsortialHoldings($id, $patron, $ids)
+    {
+        return $this->combineArraysOfAssociativeArrays(
+            'getConsortialHoldings',
+            func_get_args(),
+            ['holdings', 'electronic_holdings']
+        );
+    }
+
+    /**
+     * Get Courses
+     *
+     * Obtain a list of courses for use in limiting the reserves list.
+     *
+     * @return array An associative array with key = ID, value = name.
+     */
+    public function getCourses()
+    {
+        return $this->defaultCall('getCourses', func_get_args());
+    }
+
+    /**
+     * Get Default Pick Up Location
+     *
+     * Returns the default pick up location
+     *
+     * @param array $patron      Patron information returned by the patronLogin
+     * method.
+     * @param array $holdDetails Optional array, only passed in when getting a list
+     * in the context of placing a hold; contains most of the same values passed to
+     * placeHold, minus the patron data. May be used to limit the pickup options
+     * or may be ignored.
+     *
+     * @return string A location ID
+     */
+    public function getDefaultPickUpLocation($patron = false, $holdDetails = null)
+    {
+        return $this->defaultCall('getDefaultPickUpLocation', func_get_args(), [], true);
+    }
+
+    /**
+     * Get Departments
+     *
+     * Obtain a list of departments for use in limiting the reserves list.
+     *
+     * @return array An associative array with key = dept. ID, value = dept. name.
+     */
+    public function getDepartments()
+    {
+        return $this->defaultCall('getDepartments', func_get_args());
+    }
+
+    /**
+     * Get Funds
+     *
+     * Return a list of funds which may be used to limit the getNewItems list.
+     *
+     * @throws ILSException
+     * @return array An associative array with key = fund ID, value = fund name.
+     */
+    public function getFunds()
+    {
+        return $this->defaultCall('getFunds', func_get_args());
+    }
+
+    /**
+     * Get Default "Hold Required By" Date (as Unix timestamp) or null if unsupported
+     *
+     * @param array $patron   Patron information returned by the patronLogin method.
+     * @param array $holdInfo Contains most of the same values passed to
+     * placeHold, minus the patron data.
+     *
+     * @return int
```

Comment:
In Aleph driver this is `int|null`, so we should standardize on that.

Reply:
I changed this here and also in the Demo driver.